cypress_test/integration_tests/desktop/impress/scrolling_spec.js

```diff
@@ -31,10 +31,10 @@ describe(['tagdesktop', 'tagnextcloud', 'tagproxy'], 'Scroll through document',
 		cy.cGet('.leaflet-layer').click('right');
 		cy.wait(1000);
 		clickOnTheCenter();
-		desktopHelper.pressKey(9,'uparrow');
+		desktopHelper.pressKey(15,'uparrow');
 		desktopHelper.assertScrollbarPosition('vertical', 0, 1);
 		desktopHelper.pressKey(18,'downarrow');
-		desktopHelper.assertScrollbarPosition('vertical', 306, 355);
+		desktopHelper.assertScrollbarPosition('vertical', 160, 200);
```

Comment:
hmm, PR title is `Move the scroll related code`.
I would expect: we do a code move, tests protect the feature so no test change is done.
Later we do second PR with behavior change if needed - separate problem solved, minimal change targeting it.

That makes us sure we didn't do a regression while "just moving the code".

The commit message of any commit doesn't explain why we needed to do any adjustments.
So it is not clear for me now. What is the difference?

Reply:
Some tests are also checking which pages are visible at the moment. I didn't need to change any of them.

Difference:
The scroll bar position calculation is improved. horizontalScrollRatio and verticalScrollRatio values were differently before. Now i fixed the calculation there.
The scroll bar position values changed after the correction. I think this is safe.

First commit message also says "We need to make the scroll section document-view agnostic." It is not all about just moving the code. This move doesn't imply a copy paste. This PR hands the scrolling to ViewLayout class. And the purpose is to use view-size and the view-position for scrolling the document. Until now, document-position was used for scrolling.